### ISSUE TITLE
onScrolled and onScrolledTop shouldn't be required

### ISSUE THREAD

**Issue by jacobjanak** [thk2b/autoscroll-react#14]
Look at this silly piece of code I had to write to get autoscroll-react to work with the event listener. I think onScrolled and onScrolledTop should be optional. It looks like some code was written to make those parameters optional but it isn't working.
```
<MoveList
    moves={moves}
    onScrolled={e => {/* just need this function to exist */}}
    onScrolledTop={e => {/* just need this function to exist */}}
/>
```

**Comment by thk2b** [comment]
The props are made optional here, I'm not sure why you had to specify the props. https://github.com/thk2b/autoscroll-react/blob/bd5dd39fcdc9f2393815db91c01fd5c695c885be/src/index.js#L31
Would you mind sharing what happens without the no-ops?

**Comment by jacobjanak** [comment]
I spent some time testing this after seeing that your example works perfectly without props and I determined it's because of how I was calling setState on my array.

This is how I originally had it:
```
this.setState(state => {
    state.moves.push(newMove)
    return state;
})
```

This way matches your examples and works perfectly:
```
this.setState({
    moves: this.state.moves.concat(newMove)
})
```

So basically it was user error. Updating the array the first way was working and correctly triggering a re-render until I introduced autoscroll-react. For some reason, adding in those empty functions fixed it. Still a mystery to me as to why empty functions or setting the state a different way fixes it...
